File: tools/llm-bot/src/main/java/org/finos/springbot/tool/llm/AbstractLLMService.java
package org.finos.springbot.tool.llm;

import org.springframework.beans.factory.InitializingBean;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.io.Resource;
import org.springframework.core.io.ResourceLoader;

import com.fasterxml.jackson.databind.ObjectMapper;

public abstract class AbstractLLMService implements LLMService, InitializingBean {


    protected Instructions instructions;

	@Autowired
	private ResourceLoader resourceLoader;

	@Override
	public void afterPropertiesSet() throws Exception {
		Resource resource = resourceLoader.getResource("classpath:instructions.json");
		ObjectMapper objectMapper = new ObjectMapper();
		Instructions instructions = objectMapper.readValue(resource.getInputStream(), Instructions.class);
		System.out.println(instructions);
        this.instructions = instructions;
	}
}

File: tools/llm-bot/src/main/kotlin/org/finos/springbot/tool/llm/AbstractLLMService.kt
package org.finos.springbot.tool.llm

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.beans.factory.InitializingBean
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.core.io.ResourceLoader

abstract class AbstractLLMService : LLMService, InitializingBean {

    protected var instructions: Instructions? = null

    @Autowired lateinit var resourceLoader: ResourceLoader

    override fun afterPropertiesSet() {
        val resource = resourceLoader.getResource("classpath:instructions.json")
        val objectMapper = ObjectMapper()
        instructions = objectMapper.readValue(resource.inputStream, Instructions::class.java)
        println(instructions)
    }
}
